src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -3048,8 +3089,15 @@ MulticopterPositionControl::task_main()
 		/* set dt for control blocks */
 		setDt(dt);
 
-		/* set default max velocity in xy to vel_max */
-		_vel_max_xy = _params.vel_max_xy;
+		/* set default max velocity in xy to vel_max
+		 * Apply estimator limits if applicable */
+		if (_local_pos.vxy_max > 0.001f) {
+			/* use the estimator specified limit and allow for a minimum of 0.3 m/s for repositioning */
+			_vel_max_xy = fmaxf(_local_pos.vxy_max, 0.3f);
+		}
+
+		/* never exceed the parameter specified maximum */
+		_vel_max_xy = fminf(_vel_max_xy, _params.vel_max_xy);
```

Comment:
@priseborough That part doesn't work. `_vel_max_xy` is initialized to `0.0f` and will stay very small.
Try to run `make posix_sitl_default gazebo` and fly in position control. The quad will barely move.

Reply:
because `_vel_sp`/`vel_norm_xy` is bound to `_vel_max_xy` https://github.com/PX4/Firmware/blob/pr-ekfPosCtrlLimits/src/modules/mc_pos_control/mc_pos_control_main.cpp#L1393

Reply:
That's strange i didn't get that result when I flew outdoors earlier, but it is definitely doing it on SITL. Will fix.